src/dbnode/storage/bootstrap/types.go

```diff
@@ -416,5 +419,44 @@ type Source interface {
 	// A bootstrapper source should only return an error should it want to
 	// entirely cancel the bootstrapping of the node, i.e. non-recoverable
 	// situation like not being able to read from the filesystem.
-	Read(ctx context.Context, namespaces Namespaces) (NamespaceResults, error)
+	Read(ctx context.Context, namespaces Namespaces, runState RunState) (NamespaceResults, error)
+}
+
+// InfoFileResultsPerShard maps shards to info files.
+type InfoFileResultsPerShard map[uint32][]fs.ReadInfoFileResult
+
+// InfoFilesByNamespace maps a namespace to info files grouped by shard.
+type InfoFilesByNamespace map[namespace.Metadata]InfoFileResultsPerShard
+
+// RunState provides a snapshot of info files for use throughout all stages of the bootstrap.
```

Comment:
nit: either `State` or `BootstrapState`, probably leaning towards `State` to avoid `bootstrap.BootstrapState` stutter

Reply:
Yeah, I thought state might be a little too general, but I guess it's scoped by the package of `bootstrap`